storm-client/src/jvm/org/apache/storm/topology/WindowPartitionCache.java

```diff
@@ -0,0 +1,142 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.storm.topology;
+
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * A loading cache abstraction for caching {@link PersistentWindowedBoltExecutor.WindowPartition}.
+ *
+ * @param <K> the key type
+ * @param <V> the value type
+ */
+public interface WindowPartitionCache<K, V> {
+
+    /**
+     * Get value from the cache or load the value.
+     *
+     * @param key the key
+     * @return the value
+     */
+    V get(K key);
+
+    /**
+     * Get value from the cache or load the value pinning it
+     * so that the entry will never get evicted.
+     *
+     * @param key the key
+     * @return the value
+     */
+    V getPinned(K key);
```

Comment:
This method name can be changed to `pinAndGet` as `getPinned` was implies getting the value if it is pinned.